service/trino-proxy/src/main/java/io/trino/proxy/AuthPreservingInterceptor.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.proxy;
+
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+import java.io.IOException;
+
+public class AuthPreservingInterceptor
+        implements Interceptor
+{
+    @Override
+    public Response intercept(Interceptor.Chain chain)
+            throws IOException
+    {
+        Request request = chain.request();
+        Response response = chain.proceed(request);
+
+        if (response.isRedirect()) {
+            String authHeader = request.header("Authorization");
```

Comment:
lowercase?

Reply:
are there some constants defined?

Reply:
Yes, used.

The get method is case insensitive in okhttp:

```
if (name.equals(namesAndValues[i], ignoreCase = true)) {
          return namesAndValues[i + 1]
        }
```